### ISSUE TITLE
Screen reader count of steps in side nav is misleading

### ISSUE THREAD

**Issue by dv297** [department-of-veterans-affairs/va.gov-team#131856]
### Description

The Side Navigation component utilizes the VADS Side Navigation component, which under the hood utilizes the USWDS Accordion component. 

Based on the current implementation, when a veteran navigates the Side Nav using VoiceOver and they have filled up to the Mental Health Statement, they get these prompts:

1. heading level 2, Related Pages Menu, expanded, button
2. Related Pages Menu, Navigation
3. List 3 items
4. Link, Step 1: Veteran Details, 1 of 3 items
5. Link, Step 2: Conditions, 2 of 3 items
6. current page, visited, link, Step 3: Mental health statement
7. Step 4: Supporting evidence

There are potentially several issues here:

* The "total" count of items is based on the interact-able elements. So even though there are 6 steps in the Side Nav, the screen reader says that it is 1 of 3 because only 3 are interact-able based on our current progress through the form.
* The current page does not indicate that it is 3 of 3.
* The remaining items do not indicate to the user how many items are left in the list.

### How severe is the problem?

- [ ] Critical - A Veteran is blocked or significantly hindered from submitting the form, data is lost, or claim and evidence test coverage is missing. Fix ASAP.
- [x] Moderate - A Veteran is not blocked, but the happy path is interrupted. Fix after critical bugs.
- [ ] Minor - Cosmetic bugs and minor language inconsistencies. Defer to sprint planning.

### Desired behavior

The screen reader should clearly outline which step the user is on, as well as how many steps there might be remaining.

### Tasks

- Update the SideNav to ensure the screen reader indicates the correct total number of steps.
  - Daniel Note - My hunch is maybe nesting these under an ordered list may help. 
- Update the SideNav to ensure the current page indicates how many steps are remaining.

### Acceptance Criteria

- [ ] Desired behavior is met
- [ ] Desired behavior is validated using screen reader
- [ ] Code reviewed and approved by the Platform Team
- [ ] Bugfix tested and reviewed by Design

### Definition of Done

- [ ] Meets acceptance criteria
- [ ] Reviewed and approved by product and/or design

### Engineering

- [ ] All tests pass
- [ ] New functionality is covered by unit tests
- [ ] Logging and monitoring are implemented (if applicable)

### Code Review & Pull Requests

- [ ] PR includes Local testing steps
- [ ] PR includes Flipper/testing state details (if applicable)
- [ ] PR includes Author's local proof of submission screenshot
- [ ] Copilot review completed and feedback addressed
- [ ] Internal reviewer approved
- [ ] Internal reviewer added local proof of submission screenshot
- [ ] Code functionality verified on Staging after merge

### Refinement Checklist!

- [ ] Added description, tasks, and acceptance criteria
- [ ] Added estimate
- [ ] Labeled with Practice Area (engineer, design, product, data science)
- [ ] Labeled with issue type and characteristics of the ticket (bug, accessibility, request, discovery, documentation, research, content, ux testing, front-end, back-end, datadog, etc.)
- [ ] Added any other relevant project fields (team, OCTO priority...)
- [ ] Added an Epic or Super Epic
